### PR TITLE
Add migration to consolidate enterprise_plan_v1 into v0

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

Followup to https://github.com/dimagi/commcare-hq/pull/30988. This includes the migration which will move software plan versions referencing enterprise_plan_v1 to reference enterprise_plan_v0, at which point we can delete enterprise_plan_v1. 

The end goal is no behavior change, and every enterprise account to be unified under the same role (enterprise_plan_v0).
### Automated test coverage
Automated tests for the `change_role_for_software_plan_version`.

### QA Plan

Can test on staging before rolling out to production

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

Intentionally kept this migration separate from code changes to avoid needing to roll this specific PR back as it wouldn't be trivial. This modifies software plan versions for specific subscriptions that reference `enterprise_plan_v1`. After the migration, we won't be able to tell which plans were on `enterprise_plan_v1`.

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by biyeun — APPROVED** [review]
fantastic! 🎉

**Comment by gherceg** [comment]
@biyeun as we discussed, I made the change for the change method to support updating inactive versions as well (with test coverage). See e25f0665ec543f7c4b9df99c3524c759784b8836

File: corehq/apps/accounting/management/commands/change_role_for_software_plan_version.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def change_role_for_software_plan_version(old_role, new_role, dry_run=False):
     except Role.DoesNotExist:
         raise NewRoleDoesNotExist
 
-    versions = SoftwarePlanVersion.objects.filter(role__slug=old_role, is_active=True)
+    versions = SoftwarePlanVersion.objects.filter(role__slug=old_role)
```
Comment (biyeun) [inline corehq/apps/accounting/management/commands/change_role_for_software_plan_version.py:41]: excellent! thanks!

**Review by biyeun — APPROVED** [review]
looks solid. thanks for making that change 🎉

**Pull request merged**